NOM_visualizations/R/NOM_R_notebook.ipynb

```diff
@@ -0,0 +1,1388 @@
+{
```

Comment:
<p>Hmmm okay then I don't really understand OmicsProcessing or Processed Sample or Sample Processing or what the differences between them are... How would you explain OmicsProcessing to non-schema fluent users? I was kind of trying to use non-technical language since they won't know the whole of the schema but I ended up just being ambiguous</p>

Reply:
OmicsProcessing is like the actualy analysis happening. Like sequencing or mass spectrometry. It has an inpute of a Material object (Ilike a Biosample or ProcessedSample) and an output of a DataObject. Just like when you sequence something you put in some DNA and get a bunch of reports (data objects). Sample Processing - I think we call it Material Processing, but they sound similar - is the sample preparation. Is it filtered, extracted, etc? In NMDC we use inputs outputs. A biosample could be an input into, say, an Extraction, that has an output of a ProcessedSample, which could then be an input into, say, a LibraryPreparation, and have an output of a ProcessedSample. And then that ProcessedSample could be the input into OmicsProcessing (aka the sequencing or mass spec or whatever)

The only study I am aware of whos samples follow the Biosample -> MaterialProcessing Step -> ProcessedSample -> OmicsProcessing is NEON. The rest go from Biosample directly to OmicsProcessing. Which according to your code is what you are doing. You don't check any of the MaterialPRocessing classes, which is probably okay, but if there are NEON samples that go through NOM, you would be missing those in your data. Not sure if we really want to go down that hill (that's what we do in the Taxonomy notebooks and its confusing and arduous).

Reply:
Thank you so much for that explanation! I'll go look at the taxonomy notebook. And correct the wording of the text. If I add a check on the omics processing dataframe to make sure all the values in `has_input` are Biosample IDs, would that answer this question?

@samobermiller fyi helpful explanations of schema from Brynn

Reply:
Yeah it probably would. Great idea!